### PR TITLE
Add a CLI option to disable the progress bar on web-ext sign

### LINKED ISSUE #2154: Add an environment variable to remove the progress bar for validating & signing the add on with the web-ext sign command
### Is this a feature request or a bug?
This is a feature request.

### What is the current behavior?
Currently when we run web-ext sign command there is a progress bar that for validating the add on and signing the addon(added images below). Don't get me wrong, the progress bar is very cool on my local terminal but on CI this causes unnecessary logging.
image
image
image

### What is the expected or desired behavior?
I would like it if there was an environment variable that I could pass into the command to remove the progress updates. I would only like to remove the progress bar updates. I believe it is still valuable for us to provide the validation results link, the output of downloaded file and the Extension ID.

### LINKED ISSUE #2429: Quiet `sign` progress bar
### Is this a feature request or a bug?

Feature request

### What is the current behavior?

I'm using web-ext programmatically inside a build script. I'm calling `sign` like this:

```typescript
import webExt from "web-ext";

await webExt.cmd.sign({
  apiKey,
  apiSecret,
  verbose: false,  // this doesn't seem to help
  channel: "unlisted",
});
```

Running this build script from an interactive terminal is fine: A progress bar is "animates".

But, **when the output stream is not an interactive terminal** (but instead something like a log file on a CI runner), this renders usually several thousands of lines of:

```
...
Signing add-on [ .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  . ]
Signing add-on [  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .]
Signing add-on [.  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  ]
Signing add-on [ .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  . ]
Signing add-on [  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .]
Signing add-on [.  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  ]
Signing add-on [ .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  . ]
Signing add-on [  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .]
Signing add-on [.  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  ]
Signing add-on [ .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  . ]
Signing add-on [  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .]
Signing add-on [.  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  ]
...
```

Looking at some sample log files, there appear to be about **20 lines/progress-bar-updates per second**.

### What is the expected or desired behavior?

The progress bar should be toggle-able with some kind of option.

From what I can tell, this output actually comes from <https://github.com/mozilla/sign-addon>. Recently, that project implemented [Made progress bar optional in Client](https://github.com/mozilla/sign-addon/pull/826), which provides the `disableProgressBar` option. See <https://github.com/mozilla/sign-addon/blob/master/src/index.js#L62>

However, web-ext is the glue to sign-addon, and does not pass through the `disableProgressBar`  option. See <https://github.com/mozilla/web-ext/blob/master/src/cmd/sign.js#L114>, where only a subset of the sign-addon options are passed.

There seem to be a few ways this could be fixed, but at the very least, `disableProgressBar` should a recognized and passed-through option to `webExt.cmd.sign`.

### Version information (for bug reports)

* **Firefox version**: `101.0`
* **Your OS and version**: `Windows 10`
* **Node**: `v18.3.0`
* **npm**: `8.11.0`
* **web-ext**: `6.8.0`

### PR DESCRIPTION
**Pull request by Cimbali**
Fixes #2429 and fixes #2154 (though with a CLI option and not an environment variable.)

Help output:
```
> web-ext sign --help
web-ext sign

Sign the extension so it can be installed in Firefox

Options:
      --version             Show version number                                                                                                               [boolean]
  -s, --source-dir          Web extension source directory.                                 [string] [required] [default: "/path/to/webexts/markdown-viewer"]
  -a, --artifacts-dir       Directory where artifacts will be saved.      [string] [required] [default: "/path/to/webexts/markdown-viewer/web-ext-artifacts"]
  -v, --verbose             Show verbose output                                                                                                               [boolean]
  -i, --ignore-files        A list of glob patterns to define which files should be ignored. (Example: --ignore-files=path/to/first.js path/to/second.js "**/*.log")
                                                                                                                                                                [array]
      --no-input            Disable all features that require standard input                                                                                  [boolean]
  -c, --config              Path to a CommonJS config file to set option defaults                                                                              [string]
      --config-discovery    Discover config files in home directory and working directory. Disable with --no-config-discovery.                [boolean] [default: true]
      --amo-base-url        Signing API URL prefix - only used with `use-submission-api`            [string] [required] [default: "https://addons.mozilla.org/api/v5/"]
      --api-key             API key (JWT issuer) from addons.mozilla.org                                                                            [string] [required]
      --api-secret          API secret (JWT secret) from addons.mozilla.org                                                                         [string] [required]
      --api-url-prefix      Signing API URL prefix                                                   [string] [required] [default: "https://addons.mozilla.org/api/v4"]
      --api-proxy           Use a proxy to access the signing API. Example: https://yourproxy:6000                                                             [string]
      --use-submission-api  Sign using the addon submission API                                                                                               [boolean]
      --id                  A custom ID for the extension. This has no effect if the extension already declares an explicit ID in its manifest.                [string]
      --timeout             Number of milliseconds to wait before giving up                                                                                    [number]
      --quiet               Quieter signing process by hiding progress bars                                                                                   [boolean]
      --channel             The channel for which to sign the addon. Either 'listed' or 'unlisted'                                                             [string]
      --amo-metadata        Path to a JSON file containing an object with metadata to be passed to the API. See https://addons-server.readthedocs.io/en/latest/topics/a
                            pi/addons.html for details. Only used with `use-submission-api`                                                                    [string]
  -h, --help                Show help                                                                                                                         [boolean]
```

Failed signing without `--quiet`, has the “Validating add-on” progress bar:
```
> web-ext sign $API_CREDS
Building web extension from /path/to/webexts/markdown-viewer
Validating add-on [...................................................................................................................................................]
Validation results: https://addons.mozilla.org/en-US/developers/upload/1a507954c8ac4453acdf6d8dd93fc311
Your add-on failed validation and could not be signed
FAIL

WebExtError: The extension could not be signed
    at file:///path/to/web-ext/lib/cmd/sign.js:152:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Program.execute (file:///path/to/web-ext/lib/program.js:265:7)
    at async file:///path/to/web-ext/bin/web-ext.js:13:1
```

Failed signing with `--quiet`, there is no “Validating add-on” progress bar:
```
> web-ext sign --quiet $API_CREDS
Building web extension from /path/to/webexts/markdown-viewer
Validation results: https://addons.mozilla.org/en-US/developers/upload/46a042fc9c5a4200b7510f3bad774c3d
Your add-on failed validation and could not be signed
FAIL

WebExtError: The extension could not be signed
    at file:///path/to/web-ext/lib/cmd/sign.js:152:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Program.execute (file:///path/to/web-ext/lib/program.js:265:7)
    at async file:///path/to/web-ext/bin/web-ext.js:13:1
```

### CODE REVIEW

**Comment by Cimbali** [comment]
Now with progress bar automatically disabled when stdin is non-TTY:
```
> cat | web-ext sign --channel=unlisted `cat ../api_creds.txt` 
Building web extension from /path/to/webexts/markdown-viewer
Validation results: https://addons.mozilla.org/en-US/developers/upload/8045e4c2190d4909abda815ac27d2bad
Your add-on failed validation and could not be signed
FAIL

WebExtError: The extension could not be signed
    at file:///path/to/web-ext/lib/cmd/sign.js:153:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Program.execute (file:///path/to/web-ext/lib/program.js:265:7)
    at async file:///path/to/web-ext/bin/web-ext.js:13:1
```

**Comment by luckyrat** [comment]
This can work alongside the `--verbose` flag, yes?

If so, perhaps it would be better called `--no-progress` rather than `--quiet` since being "verbose" and "quiet" at the same time is potentially confusing.

**Comment by Cimbali** [comment]
The option can certainly be renamed `--no-progress` or `--no-progress-bars` or anything deemed appropriate.

I think @rpl (probably) should review at some point?

**Comment by willdurand** [comment]
Since the CLI flag is passed to a `disableProgressBar` attribute, I would probably rename the flag to `--disable-progress-bar`.

File: src/cmd/sign.js [hunk 1]
```diff
@@ -197,6 +200,7 @@ export default function sign(
           ...signSubmitArgs,
           apiUrlPrefix,
           apiProxy,
+          disableProgressBar: quiet,
```
Comment (willdurand) [inline src/cmd/sign.js:203]: We'd need a test case to make sure that the value of the CLI flag is indeed passed to `signAddon()`.

**Comment by Cimbali** [comment]
Updated as requested @willdurand

File: src/program.js [hunk 2]
```diff
@@ -613,7 +613,7 @@ Example: $0 --help run.
           describe: 'Number of milliseconds to wait before giving up',
           type: 'number',
         },
-        quiet: {
+        'disable-progress-bar': {
           describe: 'Quieter signing process by hiding progress bars',
```
Comment (willdurand) [inline src/program.js:617]:
```suggestion
          describe: 'Disable the progress bar in sign-addon',
```

**Review by willdurand — COMMENTED** [review]
Looking good, thanks.

**Comment by willdurand** [comment]
I closed and re-opened this PR on purpose to trigger CI jobs (don't ask...)

**Comment by willdurand** [comment]
One more thing to fix apparently:

```
> web-ext@7.5.0 prettier-ci /home/circleci/web-ext
> prettier --list-different '**' || (echo '

This failure means you did not run `npm run prettier-dev` before committing

' && exit 1)

tests/unit/test-cmd/test.sign.js


This failure means you did not run `npm run prettier-dev` before committing
```

**Comment by Cimbali** [comment]
Done. The prettier-dev step might be worth mentioning in the README.

**Pull request merged**